File: 4344.c
#include<stdio.h>

float averageOfScores(int *array, int sizeOfArray);
float percentageAboveAverage(int *array, int sizeOfArray, float averageScore);

int main(){
    int numOfCases;
    int numOfSubjects;
    
    scanf("%d\n", &numOfCases);
    
    for(int i = 0; i < numOfCases; i++){
        scanf("%d\n", &numOfSubjects);
        int scoreArray[numOfSubjects];
        
        float averageScore = 0;
        float result = 0;
        
        for(int j = 0; j < numOfSubjects; j++){
            scanf("%d", &scoreArray[j]);    
        }    
        
        //size를 변수로 주고 해결
        averageScore = averageOfScores(scoreArray, sizeof(scoreArray)/sizeof(int));
        result = percentageAboveAverage(scoreArray, sizeof(scoreArray)/sizeof(int), averageScore);
        
        printf("%.3f%%\n", result);
    }
    return 0;
}
        //함수내에서 size를 해결하려하면 size가 2로 고정되는 문제가 생김. 이유는 모르겠다.
float averageOfScores(int *array, int sizeOfArray){
    int sum = 0;
    for(int i = 0; i < sizeOfArray; i++)    sum = sum + array[i];
    return (float)sum/sizeOfArray;
        //형변환은 한번씩만 해줘도 될듯.
}

float percentageAboveAverage(int array[], int sizeOfArray, float averageScore){
    int averageAboveCounter = 0;
    for(int i = 0; i < sizeOfArray; i++)    if((float)array[i] > averageScore)    averageAboveCounter++;
    return (float)(averageAboveCounter*100)/sizeOfArray;
}

File: 4344.java
import java.util.Scanner;

public class Main{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int numOfCases = sc.nextInt();
        for(int i = 0; i < numOfCases; i++){
            int numOfSubjects = sc.nextInt();
            int[] scoreArray = new int[numOfSubjects];
            float averageScore = 0;
            float result = 0;
            //if you don't initialize those two variables, value may affected by other cases which is not cool with this
            for(int j = 0; j < numOfSubjects; j++){
                scoreArray[j] = sc.nextInt();
            }
            averageScore = averageScore(scoreArray, scoreArray.length);
            System.out.format("%.3f%%\n", aboveAverageCounter(scoreArray, scoreArray.length, averageScore));
            //if you wanna find out array's size, use .length and in case of String, .length()
            //you can print under point position number by using function like above
        }
    }
    public static float averageScore(int[] array, int sizeOfArray){
        int sum = 0;
        for(int i = 0; i < sizeOfArray; i++)    sum = sum + array[i];
        return (float)sum/sizeOfArray;
    }
    public static float aboveAverageCounter(int[] array, int sizeOfArray, float averageScore){
        int averageAboveCount = 0;
        for(int i = 0; i < sizeOfArray; i++)    if(array[i] > averageScore) averageAboveCount++;
        return (float)(averageAboveCount*100)/sizeOfArray;
    }
}



